src/app/shared/services/dynamic-data/dynamic-data.service.ts

```diff
@@ -223,4 +237,18 @@ export class DynamicDataService extends AsyncServiceBase {
     }
     return schema;
   }
+
+  evaluateItemIndexString(targetIndexString: string, currentIndex: number) {
+    if (isNaN(Number(targetIndexString))) {
+      try {
+        // HACK use keyword "item_index" to refer to index of current item in loop
+        const jsExpression = targetIndexString.replace("item_index", currentIndex.toString());
```

Comment:
nit(blocking)
Not a huge fan of this arbitrary lookup string. If using some sort of keyword/variable replacement it should ideally be prefixed in a clear way (sometimes we use '@', sometimes '_')

In this case all items already have an _index property, so I think you should be able to refer to @item._index + 1 and it be calculated without any need for this function at all